File: python/calculate_mode.py
def calculate_mode(original_list):
    new_object = {}
    final_mode = []

    for i in range(0, len(original_list)):
        if original_list[i] in new_object:
            new_object[original_list[i]] += 1
        else:
            new_object[original_list[i]] = 1
    # print(new_object)

    all_values = list(new_object.values())
    # print(all_values)
    for test in new_object:
        if new_object[test] == max(all_values):
            final_mode.append(test)
    # print(final_mode)
# calculate_mode([10,3,4,2,3,2])
    return final_mode
print(calculate_mode(["Who", "Hello", "Who"]))
    

File: js/calculateMode.js
exports.calculateMode = (originalList) => {

    let newObject = {};
    let finalMode = [];

    // Creates an object
    for (let i = 0; i < originalList.length; i++){
       if (newObject.hasOwnProperty(originalList[i])){
           newObject[originalList[i]]++
       }else{
        newObject[originalList[i]] = 1
       }
    }
    let finalValues = Object.values(newObject);
    // console.log(newObject);

    // Checks if the props value is also the highest value in the array
    for (let prop in newObject){
        if (newObject[prop] == Math.max.apply(0, finalValues)){
            finalMode.push(prop);
        } 
    }
    for (let k = 0; k < finalMode.length; k++){
        if (isNaN(finalMode[k]) == true) continue;
        finalMode[k] = parseInt(finalMode[k]);
    }
    console.log(finalMode);
}
// calculateMode([1,3,4,4,2,1]);
// calculateMode([2,2,1,7,10,12,13]);
// calculateMode([1,10,4,5,3,15,1,8,7,15]);
// calculateMode(["Who", "Hello", "Who"]);